docs: add note in upgrade docs about new Query Id generation (#3559)
confluentinc/ksql: docs/installation/upgrading.rst

Upgrade one KSQL server at a time (i.e. rolling restart). The remaining KSQL servers should have sufficient spare
capacity to take over temporarily for unavailable, restarting servers.

Upgrading to KSQL 5.4
-----------------------------------

Notable changes in 5.4:

* KSQL Server

    * Query Id generation

        * This version of KSQL includes a change to how query ids are generated for Persistent Queries
          (INSERT INTO/CREATE STREAM AS SELECT/CREATE TABLE AS SELECT). Previously, query ids would be incremented
          on every successful Persistent Query created. New query ids are now generated by taking the Kafka record

>           offset that corresponds to the query creating command and using that as the query id.

Reviewer: ```suggestion
          on every successful Persistent Query created. New query ids use the offset of the query creating command in the KSQL command topic.
```

>           In order to prevent inconsistent query ids, you should refrain from creating new Persistent Queries while

Reviewer 2: ```suggestion
          In order to prevent inconsistent query ids, don't create new Persistent Queries while
```
